python/ray/data/_internal/planner/plan_read_op.py

```diff
@@ -31,13 +30,7 @@ def get_input_data() -> List[RefBundle]:
                         # TODO(chengsu): figure out a better way to pass read
                         # tasks other than ray.put().
                         ray.put(read_task),
-                        BlockMetadata(
-                            num_rows=1,
-                            size_bytes=len(cloudpickle.dumps(read_task)),
-                            schema=None,
-                            input_files=[],
-                            exec_stats=None,
-                        ),
+                        read_task.get_metadata(),
```

Comment:
Can you make sure this calls `cleaned_metadata(read_task)` in legacy_compat.py to implement the same logic?